rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/executor/BakePoller.groovy

```diff
@@ -144,18 +154,43 @@ class BakePoller implements ApplicationListener<ContextRefreshedEvent> {
 
   void updateBakeStatusAndLogs(String statusId) {
     BakeStatus bakeStatus = executor.updateJob(statusId)
+    Id bakesId = registry.createId('bakes')
 
     if (bakeStatus) {
       if (bakeStatus.state == BakeStatus.State.COMPLETED) {
         completeBake(statusId, bakeStatus.logsContent)
-      }
 
-      bakeStore.updateBakeStatus(bakeStatus)
+        bakeStore.updateBakeStatus(bakeStatus)
+
```

Comment:
I think if you put a "Id bakesId" in the outer scope and determine the ID you want to update here, then you can make this if block just determine the id, then factor out the  bakeStore.updateBakeStatus and registry.timer(bakesId) after the if block in one place that all cases flow through.